application/application.go

```diff
@@ -51,12 +51,42 @@ type PlayedItem struct {
 
 type CastMessageFunc func(*pb.CastMessage)
 
-type Application struct {
-	conn  *cast.Connection
+type Application interface {
+	SetConn(conn cast.Connection)
+	SetIface(*net.Interface)
+	SetDebug(bool)
+	SetCacheDisabled(bool)
+	SetConnectionRetries(int)
+	SetServerPort(int)
+	Start(addr string, port int) error
+	Close(stopMedia bool) error
+	LoadApp(appID, contentID string) error
+	Status() (*cast.Application, *cast.Media, *cast.Volume)
+	Update() error
+	Pause() error
+	Unpause() error
+	Stop() error
+	StopMedia() error
+	Seek(value int) error
+	SeekFromStart(value int) error
+	SeekToTime(value float32) error
+	Load(filenameOrUrl, contentType string, transcode, detach, forceDetach bool) error
+	QueueLoad(filenames []string, contentType string, transcode bool) error
+	Transcode(command string, contentType string) error
+	Next() error
+	Previous() error
+	SetVolume(value float32) error
+	SetMuted(value bool) error
+	Slideshow(filenames []string, duration int, repeat bool) error
+	AddMessageFunc(f CastMessageFunc)
+	PlayedItems() map[string]PlayedItem
+	PlayableMediaType(filename string) bool
+}
+
+type application struct {
```

Comment:
I don't see the difference between having the underlying application public as a Application struct and using an interface instead of a Application interface since the Application used to have unexported struct field.
Correct me if i'm wrong?

Reply:
Yeah, so the Application interface is supposed to be a set of common functions to run the application by hiding the underlying implementation details (usually an interface isn't just for exposing underlying fields as public). However there is situations where you may need access to the underlying struct to change behaviour or functionality in a way that isn't supported in all the implementations of the Application interface.